docs/utilities/feature_flags.md

```diff
@@ -1,57 +1,366 @@
+                "default": false,
+                "rules": {
+                  "customer tier equals premium": {
+                    "when_match": true,
+                    "conditions": [
+                      {
+                        "action": "EQUALS",
+                        "key": "tier",
+                        "value": "premium"
+                      }
+                    ]
+                  }
+                }
+              },
+              "feature2": {
+                "default": true
+              }
+          }
+      ContentType: 'application/json'
+
+  ConfigDeployment:
+    Type: AWS::AppConfig::Deployment
+    Properties:
+      ApplicationId: !Ref FeatureStoreApp
+      ConfigurationProfileId: !Ref FeatureStoreConfigProfile
+      ConfigurationVersion: !Ref HostedConfigVersion
+      DeploymentStrategyId: "AppConfig.AllAtOnce"
+      EnvironmentId: !Ref FeatureStoreDevEnv
+```
+
+The `Content` parameter is a json structure of the feature flags and rules.
+
+TODO: add steps to create new version and new deployment for the config
+
+TODO: add CDK example
+
+### Use feature flag store
+
+After you have created and configured `AppConfigStore` and added your feature configuraiton you can use the feature
+flags in your code:
+
+=== "app.py"
+
+    ```python
+    app_config = AppConfigStore(
+        environment="dev",
+        application="product-catalogue",
+        name="features"
+    )
+    feature_flags = FeatureFlags(store=app_config)
+    ctx = {"username": "lessa", "tier": "premium", "location": "NL"}
+
+    has_premium_features: bool = feature_flags.evaluate(name="premium_features",
+                                                        context=ctx,
+                                                        default=False)
+    ```
+
+=== "features.json"
+
+    ```json
+    {
+      "premium_features": {
+        "default": false,
+        "rules": {
+          "customer tier equals premium": {
+            "when_match": true,
+            "conditions": [
+              {
+                "action": "EQUALS",
+                "key": "tier",
+                "value": "premium"
+              }
+            ]
+          }
+        }
+      }
+    }
+    ```
+
+### Evaluating a single feature flag
+
+To fetch a single feature, setup the `FeatureFlags` instance and call the `evaluate` method.
+
+=== "app.py"
+
+    ```python
+      feature_flags = FeatureFlags(store=app_config)
+
+      new_feature_active: bool = feature_flags.evaluate(name="new_feature",
+                                                          default=False)
+    ```
+
+=== "features.json"
+
+    ```json
+      {
+        "new_feature": {
+          "default": true
+        }
+      }
+    ```
+
+In this example the feature flag is **static**, which mean it will be evaluated without any additional context such as
+user or location. If you want to have **dynamic** feature flags that only works for specific user group or other contex
```

Comment:
i just call these "smart flags", they change values according to session input

Reply:
Most of the blogs or articles on the feature flag use the terminology static/dynamic, and I think at this point it would create more confusion for the users to map this meaning between different words.

Reply:
Resolving as I've just improved this section to make it crispier. We'll keep `static` and `dynamic` flags to reduce cognitive load on customers having to learn new terminologies.